Model/Api/ShippingMethods.php

```diff
@@ -343,7 +346,7 @@ public function getShippingMethods($cart, $shipping_address)
 
             /** @var \Magento\Quote\Model\Quote $parentQuote */
             $parentQuote = $this->getQuoteById($cart['order_reference']);
-            if ($this->couponInvalidForShippingAddress($parentQuote->getCouponCode(), $quote)){
+            if ($this->couponInvalidForShippingAddress($parentQuote->getCouponCode(), $this->quote)){
```

Comment:
Can we avoid here transfer `$this->quote` parameter?
Can we safely call `$this->quote` directly in protected method?

Reply:
Makes sense